src/test/integration/tests/upgrade/Prooftra.t.sol

```diff
@@ -0,0 +1,131 @@
+        staker.startCheckpoint();
+
+        // 3. Pause, Fork, and Upgrade
+        _pauseForkAndUpgrade(PECTRA_FORK_TIMESTAMP);
+
+        // 4. Set Pectra Fork Timestamp & unpause
+        _setTimestampAndUnpause();
+
+        // 5. Complete in progress checkpoint
+        staker.completeCheckpoint();
+        check_CompleteCheckpoint_State(staker);
+    }
+
+    function test_VerifyWC_Fork_EarnToPod_StartCP_CompleteCP(uint24 _rand) public rand(_rand) {
+        // Initialize state
+        (User staker, ,) = _newRandomStaker();    
+        (uint40[] memory validators, ,) = staker.startValidators();
+        beaconChain.advanceEpoch_NoRewards(); 
+
+        // 1. Verify validators' withdrawal credentials
+        staker.verifyWithdrawalCredentials(validators);
+
+        // 2. Fork to Pectra
+        BeaconChainMock_PectraForkable(address(beaconChain)).forkToPectra(PECTRA_FORK_TIMESTAMP);
+
+        // 3. Upgrade EigenPodManager & EigenPod
+        _upgradeEigenLayerContracts();
+
+        // 4. Advance epoch, generating consensus rewards and withdrawing anything over Max EB
+        // Not: Nothing is withdrawn because all validators were created with 32 ETH
+        beaconChain.advanceEpoch();
+        uint64 expectedEarnedGwei = uint64(validators.length) * beaconChain.CONSENSUS_REWARD_AMOUNT_GWEI();
+
+        // 5. Start a checkpoint
+        staker.startCheckpoint();
+        check_StartCheckpoint_State(staker);
+
+        // 6. Complete in progress checkpoint
+        staker.completeCheckpoint();
+        check_CompleteCheckpoint_EarnOnBeacon_State(staker, expectedEarnedGwei);
+    }
+
+    function _pauseForkAndUpgrade(uint64 pectraForkTimestamp) internal {
+        // 1. Pause starting checkpoint, completing, and credential proofs
+        cheats.prank(pauserMultisig);
+        eigenPodManager.pause(
+            2 ** PAUSED_START_CHECKPOINT |
+            2 ** PAUSED_EIGENPODS_VERIFY_CREDENTIALS
+        );
+
+        // 2. Fork to Pectra
+        BeaconChainMock_PectraForkable(address(beaconChain)).forkToPectra(pectraForkTimestamp);
+
+        // 3. Upgrade EigenPodManager & EigenPod
+        _upgradeEigenLayerContracts();
+
+        // 4. Set proof timestamp setter to operations multisig
+        cheats.prank(eigenPodManager.owner());
+        eigenPodManager.setProofTimestampSetter(address(operationsMultisig));
+    }
+
+    function _setTimestampAndUnpause() internal {
+        // 1. Set Timestamp
+        cheats.startPrank(eigenPodManager.proofTimestampSetter());
+        eigenPodManager.setPectraForkTimestamp(
+            BeaconChainMock_PectraForkable(address(beaconChain)).pectraForkTimestamp()
+        );
+        cheats.stopPrank();
+
+        // 2. Randomly warp to just after the fork timestamp
+        // If we do not warp, proofs will be against deneb state
+        if (_randBool()) {
+            // If we warp, proofs will be against electra state
+            cheats.warp(block.timestamp + 1); 
+        }
```

Comment:
i'm pretty sure this is testing against electra regardless of whether you warp or not